tests/add-action-when-function.ava.js

```diff
@@ -0,0 +1,229 @@
+import fs from 'fs';
+import path from 'path';
+import AvaTest from './_base-ava-test';
+
+/**
+ * We are creating files in each test,
+ * but Ava runs tests in parallel by default,
+ * which can make the tests unreliable,
+ * so we need to implement the these tests with test.serial.
+ *
+ * Also we need to clean before each test, b/c we are creating files in each test.
+ * (AvaTest only cleans once before and after the test file)
+ */
```

Comment:
Thank you for leaving this here to explain. 💯

Reply:
I should probably copy that into the other test files that need to run serially.